### PR TITLE
Bugfix: hide artifacts when setting changes

### PR DESCRIPTION
**Pull request by brandonreid**

### CODE REVIEW

**Comment by pleek91** [comment]
@brandonreid  For things like edges we actually remove them rather than just hiding them. Is there a specific reason you're hiding them for this?

**Comment by brandonreid** [comment]
@pleek91 Updated to remove artifacts and events when either are hidden.

Originally, given that these are rendered in multiple places (root run, on subflows, on nodes) it felt like less places to maintain hiding things to use visibility on the element's factory rather than each parent checking and removing. After trying it though, it actually cleans up some things for the parents and seems better, nice to not have hidden things on the stage as well.

File: src/factories/node.ts [hunk 1]
```diff
@@ -119,8 +117,18 @@ export async function nodeContainerFactory(node: RunGraphNode) {
...
 
   function createArtifactsContainer(): void {
-    artifactsContainer = new Container()
-    container.addChild(artifactsContainer)
+    if (layout.isTemporal() && !settings.disableArtifacts) {
+      if (!artifactsContainer) {
+        artifactsContainer = new Container()
+      }
+
+      container.addChild(artifactsContainer)
```
Comment (brandonreid) [inline src/factories/node.ts:125]: I did double check this. Adding something to a container that has already been added won't add a duplicate instance or anything. Pixi will shift it to the bottom of the children list, nbd though as we're generally assigning z-indexes for these anyway. Looked around to make sure toggling doesn't shift anything behind other things unintentionally and all looks good.

**Review by pleek91 — APPROVED** [review]
LGTM!

File: src/factories/flowRunArtifact.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ export async function flowRunArtifactFactory<T extends ArtifactFactoryOptions>(o
     const centeredX = x - (element.width - selectedOffset) / 2
     const y = application.screen.height
       - (element.height - selectedOffset)
-      - (flowHasEvents ? eventTargetSize : flowStateSelectedBarHeight)
+      - (flowHasEvents && !settings.disableEvents ? eventTargetSize : flowStateSelectedBarHeight)
```
[the remark this answers is not in the document]
Comment (brandonreid) [reply src/factories/flowRunArtifact.ts]: I appreciate all the nits dood. Updated. 🙏

**Pull request merged**